### PR TITLE
Si 27

### PR DESCRIPTION
**Pull request by jeanhall**

### CODE REVIEW

File: Zone5/Transport/RequestEndpoint.swift [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import Foundation
 
-/// Protocol defining the requirements for endpoints accepted when generating requests for the API.
-public protocol RequestEndpoint {
+/// Protocol defining internal endpoints accepted when generating requests for the API.
+/// These endpoints contain relative uris and need to be wrapped by a RequestEndpoint to add server baseURL
+public protocol InternalRequestEndpoint {
```
Comment (johnkeller101) [inline Zone5/Transport/RequestEndpoint.swift:5]: Should this be private?

File: Zone5/Transport/StringEncodedBody.swift [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 import Foundation
 
-struct StringEncodedBody: JSONEncodedBody {
+public struct StringEncodedBody: JSONEncodedBody {
```
Comment (johnkeller101) [inline Zone5/Transport/StringEncodedBody.swift:11]: Does this need to be public? Ride does not need this struct.

**Review by johnkeller101 — CHANGES_REQUESTED** [review]
All the changes look great to me. I added a boolean for debug logging, and added a more native logging solution with a zone 5 prefix. I have not tested the multipart form data either, but I figure that should be a similar implementation.

File: Zone5/Transport/RequestEndpoint.swift (hunk 1 above)
Comment (jeanhall) [reply Zone5/Transport/RequestEndpoint.swift:5]: If i make that internal or private I get "Public protocol cannot refine an internal protocol" for the RequestEndpoint (which extends it.

The ExternalView funcs all take RequestEndpoint so you can't incorrectly pass an InternalRequestEndpoint.

File: Zone5/Transport/StringEncodedBody.swift (hunk 2 above)
Comment (jeanhall) [reply Zone5/Transport/StringEncodedBody.swift:11]: yes. If you have a custom request that takes a simple string as the post body you can use this object.

**Pull request merged**